.circleci/config.yml

```diff
@@ -15,21 +14,34 @@ jobs:
           POSTGRES_PASSWORD: postgres
           POSTGRES_USER: postgres
           POSTGRES_DB: aerogear_mobile_metrics
-      
-      # Specify service dependencies here if necessary
-      # CircleCI maintains a library of pre-built images
-      # documented at https://circleci.com/docs/2.0/circleci-images/
-      # - image: circleci/postgres:9.4
-
     #### TEMPLATE_NOTE: go expects specific checkout path representing url
     #### expecting it in the form of
     ####   /go/src/github.com/circleci/go-tool
     ####   /go/src/bitbucket.org/circleci/go-tool
     working_directory: /go/src/github.com/aerogear/aerogear-metrics-api
     steps:
-      - checkout
       - run: go get github.com/mattn/goveralls
       - run: go get -u github.com/golang/dep/cmd/dep
+      - checkout
       - run: dep ensure
       - run: make test-integration-cover
-      - run: /go/bin/goveralls -coverprofile=coverage-all.out -service=circle-ci -repotoken=$COVERALLS_TOKEN
\ No newline at end of file
+      - run: /go/bin/goveralls -coverprofile=coverage-all.out -service=circle-ci -repotoken=$COVERALLS_TOKEN
+  release:
+    filters:
+      # execute only on semver-style tags (vX.X.X)
+      tags:
+        only: /^v\d\.\d\.\d/
+      branches:
+        # needs to explicitly ignore all branches else gets run every time
+        ignore: /.*/
```

Comment:
Will this release workflow get kicked off on master? It's not too big a deal if we're not 100% sure. We can do some trial runs when this branch is merged.